src/fireedge/src/modules/components/Tables/Enhanced/WrapperRow.js

```diff
@@ -0,0 +1,211 @@
+ * ------------------------------------------------------------------------- */
+import { TABLE_VIEW_MODE } from '@ConstantsModule'
+import { useGeneral } from '@FeaturesModule'
+import EnhancedTableStyles from '@modules/components/Tables/Enhanced/styles'
+import { Checkbox, Grid, TableCell, TableRow, useTheme } from '@mui/material'
+import get from 'lodash.get'
+import PropTypes from 'prop-types'
+import { ReactElement, memo, useMemo } from 'react'
+
+/**
+ * @param {object} props - Props
+ * @returns {ReactElement} Generic Row
+ */
+const RowStyle = memo(
+  ({
+    original = {},
+    value = {},
+    onClickLabel,
+    onDeleteLabel,
+    globalErrors,
+    headerList = [],
+    className,
+    rowDataCy = '',
+    isSelected = false,
+    toggleRowSelected = () => {},
+    onClick: onClickRow = () => {},
+    enabledFullScreen = false,
+    singleSelect = false,
+    ...props
+  }) => {
+    const { ID = '' } = original
+    const theme = useTheme()
+    const styles = useMemo(
+      () =>
+        EnhancedTableStyles({
+          ...theme,
+        }),
+      [theme]
+    )
+
+    const handleChange = (event) => {
+      event?.stopPropagation()
+      toggleRowSelected(event.target.checked)
+    }
+
+    return (
+      <TableRow
+        data-cy={`list-${rowDataCy}-${ID}`}
+        {...props}
+        className={`${styles.row} ${className}`}
+      >
+        {!singleSelect && (
+          <TableCell>
+            <Checkbox checked={isSelected} onChange={handleChange} />
+          </TableCell>
+        )}
+
+        {headerList.map(({ id, accessor }) => {
+          switch (typeof accessor) {
+            case 'string':
+              return (
+                <TableCell
+                  onClick={onClickRow}
+                  key={id}
+                  className={styles.cell}
+                >
+                  {get(original, accessor)}
+                </TableCell>
+              )
+            case 'function':
+              return (
+                <TableCell
+                  onClick={onClickRow}
+                  key={id}
+                  className={styles.cell}
+                >
+                  {accessor(original, value)}
+                </TableCell>
+              )
+            default:
+              return ''
+          }
+        })}
+      </TableRow>
+    )
+  },
+  (prev, next) => prev.className === next.className
+)
+
+RowStyle.propTypes = {
+  original: PropTypes.object,
+  value: PropTypes.object,
+  onClickLabel: PropTypes.func,
+  onDeleteLabel: PropTypes.func,
+  globalErrors: PropTypes.array,
+  headerList: PropTypes.oneOfType([PropTypes.array, PropTypes.bool]),
+  rowDataCy: PropTypes.string,
+  className: PropTypes.string,
+  isSelected: PropTypes.bool,
+  toggleRowSelected: PropTypes.func,
+  onClick: PropTypes.func,
+  enabledFullScreen: PropTypes.bool,
+  singleSelect: PropTypes.bool,
+}
+
+RowStyle.displayName = 'RowStyle'
+
+const CardWrapper = memo(
+  ({ children }) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).